allow mysql_* modules to get socket connection information from my.cnf

mpeters opened the pull request:
This treats socket just like the other connection information (user and passwd) and gets them from my.cnf if it can, when it's not explicitly passed as an option.

This is based on an earlier conversation on the ansible list: https://groups.google.com/d/msg/ansible-project/npSwZM_2OFU/9D4SulSfpi8J

Commit:
allow mysql_* modules to get socket connection information from my.cnf like it does for user and passwd

jimi-c commented:
This looks really good, but there is a lot of repetition. I believe it's time we split a lot of this functionality into a new `module_utils/mysql.py` snippet that could be included in each of the mysql_\* modules as necessary. Would you be interested in working on that? If not, we can get this in queue for testing and split it out later.

mpeters commented:
@jimi-c yeah, I can definitely do that. I also noticed the need for a refactoring, but didn't want that need to hold up getting this PR in. Do you want to go ahead and merge this PR first? Or would you like my refactor to be a part of this PR? Or do you want me to do a refactor first as part of a different PR and then rework this PR to use that refactor?

jimi-c commented:
@mpeters go ahead and update this PR with that change, and we'll merge it in at once. This already touches most of the same mysql_ modules that the refactor would as well.

Commit:
Refactor mysql_* modules to load use new module_utils/mysql.py
mycnf_options helper routine added to pull user, password and socket
from MySQL's my.conf file to be used by mysql_* modules when not
explicitly passed.

mpeters commented:
@jimi-c Can you please review my latest changes that include this refactor that you recommended. Thanks

muffl0n commented:
I also worked on improving how config files are read in #6295. Unfortunately there has not been a decision on this yet.

mpdehaan commented:
Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

On September 26, 2014, due to enormous levels of contribution to the project Ansible decided to reorganize module repos, making it easier
for developers to work on the project and for us to more easily manage new contributions and tickets.

We split modules from the main project off into two repos, http://github.com/ansible/ansible-modules-core and http://github.com/ansible/ansible-modules-extras

If you still would like this pull request merged, we will need your help making this target the new repo.  If you do not take any action, this
pull request unfortunately cannot be applied.

We apologize that we are not able to make this transition happen seamlessly, though this is a one-time change and your help is greatly appreciated -- 
this will greatly improve velocity going forward.

Both sets of modules will ship with Ansible, though they'll receive slightly different ticket handling.

To locate where a module lives between 'core' and 'extras'
- Find the module at http://docs.ansible.com/list_of_all_modules.html
- Open the documentation page for that module
- If the bottom of the docs say "This is an extras module", submit your ticket to https://github.com/ansible/ansible-modules-extras
- Otherwise, submit your pull request to update the existing module to https://github.com/ansible/ansible-modules-core
- Note that python modules in ansible now also end in ".py" and this extension is required for new contributions.
- action_plugins (modules with server side components) still live in the main repo.  If your pull request touches both, which should be
   exceedingly rare, submit two new pull requests and make sure to mention the links to each other in the comments.

Otherwise, if this is a new module:
- Submit your pull request to add a module to https://github.com/ansible/ansible-modules-extras

It may be possible to re-patriate your pull requests automatically, one user-submitted approach for advanced git users
has been suggested at https://gist.github.com/willthames/afbaaab0c9681ed45619

Additionally, should you need more help with this, you can ask questions on:
- the development mailing list: https://groups.google.com/forum/#!forum/ansible-devel

Thanks you very much!
